tracer/src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -66,7 +66,13 @@ static IEnumerable<MethodInfo> GetBaseMethods(Type baseType)
             {
                 foreach (MethodInfo method in baseType.GetMethods(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic))
                 {
-                    // Avoid proxying object methods like ToString(), GetHashCode()
+                    if (method.Name == "ToString" && method.ReturnType == typeof(string) && method.GetParameters().Length == 0)
```

Comment:
I think this means that we don't have a way to _not_ proxy `ToString()`. Probably not an issue, not sure it's worth trying to support that. Just wanted to mention it.

Reply:
I’m having a hard time thinking on which scenario we will need the proxy ToString and not the target one.

Reply:
My main thought was reverse duck typing, but I'm not that applies here 🤔